node/peers.js

```diff
@@ -199,22 +222,25 @@ function choosePeer(req) {
 
     var threshold = self.peerScoreThreshold;
 
-    var selectedPeer = null;
-    var selectedScore = 0;
-    for (var i = 0; i < hosts.length; i++) {
-        var hostPort = hosts[i];
-        var peer = self._map[hostPort];
-        if (!req || !req.triedRemoteAddrs[hostPort]) {
-            var score = peer.state.shouldRequest(req);
-            var want = score > threshold &&
-                       (selectedPeer === null || score > selectedScore);
-            if (want) {
-                selectedPeer = peer;
-                selectedScore = score;
-            }
-        }
+    if (self._heap.scores[0] <= threshold) { // TODO: why inclusive?
+        return null;
     }
-    return selectedPeer;
+
+    var peer = self._heap.items[0];
```

Comment:
Please add a check for `while (req.triedRemoteAddrs[peer.hostPort]) { peer = selectNext() }`